provisioning_templates/iPXE/preseed_default_ipxe.erb

```diff
@@ -8,16 +8,25 @@ oses:
 - Ubuntu
 %>
 <% if @host.operatingsystem.name == 'Debian' -%>
-<% keyboard_params = "auto=true domain=#{@host.domain}" -%>
+  <%- keyboard_params = "auto=true domain=#{@host.domain}" -%>
 <% else -%>
-<% keyboard_params = 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us' -%>
+  <%- keyboard_params = 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us' -%>
 <% end -%>
+
+<% subnet = @host.subnet -%>
+<% if subnet.nil? || (subnet.respond_to?(:dhcp_boot_mode?) && subnet.dhcp_boot_mode?) -%>
```

Comment:
Ditto, we should not be storing any other object than Subnet in `@host.subnet`. That sounds like a bug to me.